cli/main.rs

```diff
@@ -87,26 +106,16 @@ impl From<StrBase> for Base {
     }
 }
 
-fn encode(base: StrBase) -> Result<(), ExitFailure> {
-    log::debug!("encoding with {}", base);
-    let mut stdin = io::stdin();
-    let mut stdout = io::stdout();
-    let mut buffer = Vec::new();
-    stdin.read_to_end(&mut buffer)?;
-    log::debug!("read {:?} from stdin", buffer);
-    let result = multibase::encode(base.into(), buffer.as_slice());
-    stdout.write_all(result.as_bytes())?;
+fn encode(base: StrBase, input: &[u8]) -> Result<()> {
+    println!("Encode {:?} with {}", input, base);
```

Comment:
Please use the logging functionality like `log::debug!()` instead of using `println!()`